sap/aibus/dar/client/inference_client.py

```diff
@@ -121,3 +121,33 @@ def do_bulk_inference(
             )
             result.extend(response["predictions"])
         return result
+
+    def create_inference_request_with_url(
+        self,
```

Comment:
Probably we should add some documentation also here.

Reply:
Now I am also wondering if it makes sense to have this method in the `InferenceClient` even if the `url` that has to be given in the `InferenceClient` constructor is ignored. Maybe we should create another class? Or we just stay with the workaround from Katja (Internal link: https://github.wdf.sap.corp/ML-Leonardo/ML-MEH-Assembly/pull/137/files). What do you think?

Reply:
@thedomdom I think it would make sense have things consistent in the E2E tests, and this method could find itself useful somewhere else should we decide to migrate all tests to the SDK (like how we did for the assembly tests). As for the URL being ignored in the `InferenceClient`, `DARClientWithSession` and `DARSession` can be modified to have the `url` argument optional (without breaking any of the existing usages) so that the `InferenceClient` can be created without passing a URL. Would that change make sense?

Reply:
There is also an issue open here: #121

Reply:
I think it is indeed a design challenge to now pass a URL top-down when it is currently constructed in a lower layer (`DARSession`) of the code. I think the current proposal is quite OK. The other design possibility would be to add a completely different InferenceWithURLClient class which would perhaps not use the `DARSession` because it does not need the functionality to construct the service URL.

However, from an API perspective, it would be hard to understand why there is a separate `InferenceWithURLClient` class. From a usability perspective, I think the proposed solution by @SreevishnuAB works very nicely!